src/device-manager/java/WeaveDeviceManager-JNI.cpp

```diff
@@ -3321,6 +3452,631 @@ WEAVE_ERROR GetClassRef(JNIEnv *env, const char *clsType, jclass& outCls)
     return err;
 }
 
+void EngineEventCallback(void * const aAppState,
+                                SubscriptionEngine::EventID aEvent,
+                                const SubscriptionEngine::InEventParam & aInParam, SubscriptionEngine::OutEventParam & aOutParam)
+{
+    switch (aEvent)
+    {
+        default:
+            SubscriptionEngine::DefaultEventHandler(aEvent, aInParam, aOutParam);
+            break;
+    }
+}
+
+void Java_nl_Weave_DataManagement_WDMClientImpl_init(JNIEnv *env, jobject self)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+
+    err = SubscriptionEngine::GetInstance()->Init(&sExchangeMgr, NULL, EngineEventCallback);
+    SuccessOrExit(err);
+
+exit:
+    return;
+}
+
+void BindingEventCallback (void * const apAppState, const nl::Weave::Binding::EventType aEvent,
+                                  const nl::Weave::Binding::InEventParam & aInParam, nl::Weave::Binding::OutEventParam & aOutParam)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+
+    Binding * binding = aInParam.Source;
+
+    WeaveLogDetail(DeviceManager, "%s: Event(%d)", __func__, aEvent);
+
+    WeaveDeviceManager * const pDeviceMgr = reinterpret_cast<WeaveDeviceManager *>(apAppState);
+
+    switch (aEvent)
+    {
+        case nl::Weave::Binding::kEvent_PrepareRequested:
+            WeaveLogDetail(DeviceManager, "kEvent_PrepareRequested");
+            err = pDeviceMgr->ConfigureBinding(binding);
+            aOutParam.PrepareRequested.PrepareError = err;
+            SuccessOrExit(err);
+            break;
+
+        case nl::Weave::Binding::kEvent_PrepareFailed:
+            err = aInParam.PrepareFailed.Reason;
+            WeaveLogDetail(DeviceManager, "kEvent_PrepareFailed: reason %d", err);
+            break;
+
+        case nl::Weave::Binding::kEvent_BindingFailed:
+            err = aInParam.BindingFailed.Reason;
+            WeaveLogDetail(DeviceManager, "kEvent_BindingFailed: reason %d", err);
+            break;
+
+        case nl::Weave::Binding::kEvent_BindingReady:
+            WeaveLogDetail(DeviceManager, "kEvent_BindingReady");
+            break;
+
+        case nl::Weave::Binding::kEvent_DefaultCheck:
+            WeaveLogDetail(DeviceManager, "kEvent_DefaultCheck");
```

Comment:
Is this intentionally falling through to the default case, or should there be a break statement? If the former, maybe add an explicit `// fallthrough` comment?

Reply:
yes, it is intentionally failing through, i am adding the comment.
        case nl::Weave::Binding::kEvent_DefaultCheck:
            WeaveLogDetail(DeviceManager, "kEvent_DefaultCheck");
            // fall through
thanks